pkg/errors/errors_test.go

```diff
@@ -23,3 +38,2006 @@ func TestErrorFind(t *testing.T) {
+			t.Parallel()
+			Expect(Maintenance(tt.args.reason)).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_MaximumAllowedInstanceReached(t *testing.T) {
+	type args struct {
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ServiceError
+	}{
+		{
+			name: "should return a new service error if ErrorMaxAllowedInstanceReached",
+			args: args{
+				reason: "Forbidden to create more instances than the maximum allowed",
+			},
+			want: New(ErrorMaxAllowedInstanceReached, "Forbidden to create more instances than the maximum allowed"),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(MaximumAllowedInstanceReached(tt.args.reason)).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_TooManyKafkaInstancesReached(t *testing.T) {
+	type args struct {
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ServiceError
+	}{
+		{
+			name: "should return a new service error if ErrorTooManyKafkaInstancesReached",
+			args: args{
+				reason: "The maximum number of allowed kafka instances has been reached",
+			},
+			want: New(ErrorTooManyKafkaInstancesReached, "The maximum number of allowed kafka instances has been reached"),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(TooManyKafkaInstancesReached(tt.args.reason)).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_Validation(t *testing.T) {
+	type args struct {
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ServiceError
+	}{
+		{
+			name: "should return a new service error if ErrorValidation",
+			args: args{
+				reason: "General validation failure",
+			},
+			want: New(ErrorValidation, "General validation failure"),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(Validation(tt.args.reason)).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_MalformedRequest(t *testing.T) {
+	type args struct {
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ServiceError
+	}{
+		{
+			name: "should return a new service error if ErrorMalformedRequest",
+			args: args{
+				reason: "Unable to read request body",
+			},
+			want: New(ErrorMalformedRequest, "Unable to read request body"),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(MalformedRequest(tt.args.reason)).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_FailedToParseSearch(t *testing.T) {
+	type args struct {
+		reason string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ServiceError
+	}{
+		{
+			name: "should return a new service error if ErrorFailedToParseSearch",
+			args: args{
+				reason: "",
+			},
```

Comment:
```suggestion
```

if nothing is provided, reason will have value of empty string, so the args can be skipped

Reply:
args can be removed altogether, if not provided in this test function